File: tests/fixtures/sample.py
class Calculator:
    """A simple calculator."""

    def __init__(self, initial=0):
        self.value = initial

    def add(self, x):
        self.value += x
        return self.value

    @staticmethod
    def version():
        return "1.0"


class Vector:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def magnitude(self):
        return (self.x**2 + self.y**2) ** 0.5


def greet(name):
    return f"Hello, {name}!"


def _private_helper(x, y):
    return x + y

File: tests/fixtures/sample.php
<?php

class Calculator
{
    private int $value;

    public function __construct(int $initial = 0)
    {
        $this->value = $initial;
    }

    public function add(int $x): int
    {
        $this->value += $x;
        return $this->value;
    }

    private function reset(): void
    {
        $this->value = 0;
    }
}

interface Drawable
{
    public function draw(): void;
    public function getName(): string;
}

trait Loggable
{
    public function log(string $msg): void
    {
        echo $msg;
    }
}

enum Status: string
{
    case Active = 'active';
    case Inactive = 'inactive';
}

function greet(string $name): string
{
    return "Hello, $name!";
}
